### ISSUE TITLE
(raghuramank) This code relies on string matching and needs to be 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#1183]
updated if unfused batch norm continues to be widely used
Matching variable names is brittle and relies on scoping
conventions. Fused batch norm folding is more robust. Support for unfused
batch norms will be deprecated as we move forward. Fused batch norms allow
for faster training and should be used whenever possible.
context contains part of the names of the tensors we are interested in:
For MobilenetV1, the context has repetitions:
MobilenetV1/MobilenetV1/Conv2d_3_depthwise
when the moving_mean tensor has the name:
MobilenetV1/Conv2d_3_depthwise/BatchNorm/moving_mean/read
To pick the correct variable name, it is necessary to ignore the repeating
header.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [3ab696e7e7e5c422acaa2fb2f3a938ce14effc9c](https://github.com/imdone/tensorflow/commit/3ab696e7e7e5c422acaa2fb2f3a938ce14effc9c) by Raghuraman Krishnamoorthi raghuramank@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/contrib/quantize/python/fold_batch_norms.py#L540-L557
